app/webpacker/components/NewcomerChecks/DuplicateCheckerHeader.jsx

```diff
@@ -1,16 +1,50 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { DateTime } from 'luxon';
 import { Button, Message } from 'semantic-ui-react';
 import { duplicateCheckerJobRunStatuses } from '../../lib/wca-data.js.erb';
 
-export default function DuplicateCheckerHeader({ lastDuplicateCheckerJobRun, run }) {
-  if (
-    lastDuplicateCheckerJobRun.run_status === duplicateCheckerJobRunStatuses.in_progress
-    || lastDuplicateCheckerJobRun.run_status === duplicateCheckerJobRunStatuses.not_started
-  ) {
+const JOB_RUN_DELAY_MINUTES = 10;
+
+function isJobRunDelayed(lastDuplicateCheckerJobRun) {
+  const jobNotStartedOrInProgress = (
+    lastDuplicateCheckerJobRun.run_status === duplicateCheckerJobRunStatuses.not_started
+    || lastDuplicateCheckerJobRun.run_status === duplicateCheckerJobRunStatuses.in_progress
+  );
+
+  return (jobNotStartedOrInProgress
+    && DateTime.now().diff(
+      DateTime.fromISO((lastDuplicateCheckerJobRun.start_time)),
+    ).as('minutes') > JOB_RUN_DELAY_MINUTES
+  );
+}
+
+export default function DuplicateCheckerHeader({
+  lastDuplicateCheckerJobRun, run, refetch,
+}) {
+  const jobRunDelayed = useMemo(
+    () => isJobRunDelayed(lastDuplicateCheckerJobRun),
+    [lastDuplicateCheckerJobRun],
+  );
+
+  if (jobRunDelayed) {
+    return (
+      <Message warning>
+        Job running longer than expected. Click &apos;Retry&apos; if it appears stuck.
+        <Button onClick={run}>Retry</Button>
+      </Message>
+    );
+  } if (lastDuplicateCheckerJobRun.run_status === duplicateCheckerJobRunStatuses.not_started) {
```

Comment:
You should either (a) turn this into an `else if` or (b) add a line break before the subsequent if.

Reply:
Okay opted for second one.